main.go

```diff
@@ -16,29 +16,356 @@
+		}
+	}
+
+	if len(repoRef) == 0 || len(repoPath) == 0 {
+		http.Error(w, "error extracting repo ref or path from certificate", http.StatusInternalServerError)
+		return
+	}
+
+	// Split repo path into owner and repo name.
+	ownerName := repoPath[:strings.Index(repoPath, "/")]
+	repoName := repoPath[strings.Index(repoPath, "/")+1:]
+
+	// Verify that the repository and branch of the cert and request are equal.
+	reqRepo := r.Header["Repository"]
+	reqBranch := r.Header["Branch"]
+	if len(reqRepo) == 0 || len(reqBranch) == 0 || reqRepo[0] != repoPath || reqBranch[0] != repoRef {
+		http.Error(w, "repository and branch of cert doesn't match that of request", http.StatusInternalServerError)
+		return
+	}
+
+	// Get workflow file from repo reference.
+	// TODO: use GITHUB_TOKEN from workflow to make the api call.
+	client := github.NewClient(nil)
+	opts := &github.RepositoryContentGetOptions{Ref: repoRef}
+	contents, _, _, err := client.Repositories.GetContents(ctx, ownerName, repoName, ".github/workflows/scorecards.yml", opts)
+	if err != nil {
+		http.Error(w, "error downloading workflow contents from repo", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+
+	workflowContent, err := contents.GetContent()
+	if err != nil {
+		http.Error(w, "error decoding workflow contents", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+
+	// Verify scorecard workflow.
+	verified := verifyScorecardWorkflow(workflowContent)
+	fmt.Println(verified)
+
+	// Save blob to GCS
+
+	// Next: badging...
+}
+
+func getScore(w http.ResponseWriter, r *http.Request) {
+	scorecardData := struct{ Score int }{Score: 1}
+	jData, err := json.Marshal(scorecardData)
+	if err != nil {
+		http.Error(w, "error marshalling struct", http.StatusInternalServerError)
+		fmt.Println(err)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(jData)
+}
+
 func main() {
-	http.HandleFunc("/", httpHandler)
 	fmt.Printf("Starting HTTP server on port 8080 ...\n")
+
+	r := mux.NewRouter().StrictSlash(true)
+	r.HandleFunc("/", homepage)
+	r.HandleFunc("/projects/{host}/{orgName}/{repoName}", getScore).Methods("GET")
+	r.HandleFunc("/verify", verifySignature).Methods("POST")
+	http.Handle("/", r)
+
 	if err := http.ListenAndServe(":8080", nil); err != nil {
 		log.Fatal(err)
 	}
 }
 
-func httpHandler(w http.ResponseWriter, r *http.Request) {
-	switch r.Method {
-	case http.MethodGet:
-		w.WriteHeader(http.StatusOK)
-		if _, err := w.Write([]byte("Hello world!" +
-			" This site is still under construction." +
-			" Please check back again later.")); err != nil {
-			log.Printf("error during Write: %v", err)
+func homepage(w http.ResponseWriter, r *http.Request) {
+	if _, err := fmt.Fprintf(w, "Hello world!!"+
+		" This site is still under construction."+
+		" Please check back again later."); err != nil {
+		log.Printf("error during Write: %v", err)
+	}
+}
+
+func verifyScorecardWorkflow(workflowContent string) bool {
```

Comment:
How about this func returns the `bool,error`, so that the caller can know as to why it failed. It would be easier to debug.


Another suggestion would instead just return the error when is false.

Reply:
Went with the second approach